app/lib/common/providers/notifiers/room_notifiers.dart

```diff
@@ -41,3 +44,82 @@ class AsyncMaybeRoomNotifier extends FamilyAsyncNotifier<Room?, String> {
     return await _getRoom();
   }
 }
+
+class RoomAvatarInfoNotifier extends FamilyNotifier<AvatarInfo, String> {
+  late ProviderSubscription<AsyncValue<String?>> _displayNameListener;
+  late ProviderSubscription<AsyncValue<MemoryImage?>> _avatarListener;
+  @override
+  AvatarInfo build(arg) {
+    final roomId = arg;
+
+    final fallback = AvatarInfo(uniqueId: roomId);
+
+    final room = ref.watch(maybeRoomProvider(roomId)).valueOrNull;
+    if (room == null) {
+      return fallback;
+    }
+
+    final displayName = ref.read(roomDisplayNameProvider(roomId)).valueOrNull;
+    final avatarData = ref.read(roomAvatarProvider(roomId)).valueOrNull;
+
+    _avatarListener = ref.listen(
+      roomAvatarProvider(roomId),
+      (previous, next) => _maybeUpdate(roomId),
+    );
+    _displayNameListener = ref.listen(
+      roomDisplayNameProvider(roomId),
+      (previous, next) => _maybeUpdate(roomId),
+    );
+
+    ref.onDispose(() {
+      _displayNameListener.close();
+      _avatarListener.close();
+    });
```

Comment:
`ref.watch` (what we had before) comes with the nasty side-effect of returning us `null` for `valueOrNull` on any type of `loading` taking place. That is pretty annoying as it a) means we are updating the internal object several times (once for loading to start and once for when the new data is available), but also that we produce annoying flickerings because we are resetting the internal value to `null` rather than from the oldValue -> newValue directly.

Using these listeners and the `maybeUpdate` which gradually only updates the features that _have new data availalbe_ (as in, aren't loading/refreshing/revalidating anymore) means we directly switch from oldValue -> newValue and do updates only in that case at all. Reducing the flickering in the UI that is currently caused on every room-update-trigger.